File: backend/models/skill.py
from pydantic import BaseModel
from typing import Optional

from .common import Scope


class SkillBase(BaseModel):
    name: str
    description: Optional[str] = ""
    content: Optional[str] = ""


class CreateSkillRequest(SkillBase):
    scope: Scope = "project"


class UpdateSkillRequest(BaseModel):
    description: Optional[str] = None
    content: Optional[str] = None
    scope: Optional[Scope] = None


class ImportSkillRequest(BaseModel):
    name: str
    content: str
    scope: Scope = "project"
    format: str = "markdown"  # "markdown" or "frontmatter"


class SkillResponse(BaseModel):
    name: str
    description: str = ""
    content: str = ""
    scope: Scope
    active: bool = True
    overridden: Optional[dict] = None
    source: Optional[str] = None
    path: Optional[str] = None


class MoveSkillRequest(BaseModel):
    target_scope: Scope

File: frontend/src/types/skill.ts
import type { Scope, OverrideInfo } from './common';

export interface Skill {
  name: string;
  description: string;
  content?: string;
  scope: Scope;
  active: boolean;
  overridden?: OverrideInfo;
  source?: string;
  path?: string;
}

export interface SkillResponse {
  name: string;
  description: string;
  content: string;
  scope: Scope;
  active: boolean;
  overridden?: Record<string, unknown> | null;
  source?: string;
  path?: string;
}

export interface CreateSkillRequest {
  name: string;
  content: string;
  description?: string;
  scope: Scope;
}

export interface UpdateSkillRequest {
  content?: string;
  description?: string;
  scope?: Scope;
}
